src/builders/AvailabilityModelBuilder.ts

```diff
@@ -0,0 +1,104 @@
+import { AvailabilityStatus, CapabilityId, PricingPer } from "@octocloud/types";
+import { PricingDataProvider } from "../dataProviders/PricingDataProvider";
+import { AvailabilityData } from "../data/AvailabilityData";
+import { AvailabilityModel } from "../models/Availability/AvailabilityModel";
+import { AvailabilityContentModel } from "../models/Availability/AvailabilityContentModel";
+import { AvailabilityPricingModel } from "../models/Availability/AvailabilityPricingModel";
+import { AvailabilityPickupModel } from "../models/Availability/AvailabilityPickupModel";
+import { addDays } from "date-fns";
+import { DateHelper } from "../helpers/DateHelper";
+import { TimeZoneDataProvider } from "../dataProviders/TimeZoneDataProvider";
+import { AvailabilityPricingModelFactory } from "../factories/AvailabilityPricingModelFactory";
+
+interface AvailabilityModelBuilderData {
+  availabilityData: AvailabilityData;
+  pricingPer?: PricingPer;
+  capabilities?: CapabilityId[];
+}
+
+const defaultPricingPer: PricingPer = PricingPer.UNIT;
+const defaultCapabilities: CapabilityId[] = [CapabilityId.Content, CapabilityId.Pricing, CapabilityId.Pickups];
+
+export class AvailabilityModelBuilder {
+  public build(builderData: AvailabilityModelBuilderData): AvailabilityModel {
+    builderData.pricingPer ??= defaultPricingPer;
+    builderData.capabilities ??= defaultCapabilities;
+
+    const availabilityData = builderData.availabilityData;
+    const date = new Date();
+    const dateAsString = DateHelper.formatForAvailability(date);
+    const time = "00:00:00";
+    const datetime = new Date(`${dateAsString}T${time}`);
+
+    const localDateTimeStart =
+      availabilityData.localDateTimeStart ??
+      DateHelper.formatAsAvailabilityId(datetime, TimeZoneDataProvider.europeLondon);
+    const localDateTimeEnd =
+      availabilityData.localDateTimeEnd ??
+      DateHelper.formatAsAvailabilityId(addDays(datetime, 5), TimeZoneDataProvider.europeLondon);
+
+    return new AvailabilityModel({
+      id: availabilityData.id ?? localDateTimeStart,
+      localDateTimeStart: localDateTimeStart,
+      localDateTimeEnd: localDateTimeEnd,
+      allDay: availabilityData.allDay ?? true,
+      available: availabilityData.available ?? true,
+      status: availabilityData.status ?? AvailabilityStatus.AVAILABLE,
+      vacancies: availabilityData.vacancies ?? 10,
+      capacity: availabilityData.capacity ?? 10,
+      maxUnits: availabilityData.maxUnits ?? 5,
+      utcCutoffAt: availabilityData.utcCutoffAt ?? DateHelper.formatToUtcDate(date),
+      openingHours: availabilityData.openingHours ?? [
```

Comment:
it's better to default it to the empty array

there are 2 types of availability:

start time - don't have opening hours
opening hours - this one does